src/alloc.f90

```diff
@@ -4,22 +4,67 @@
   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 
   ! X-pencil real arrays
-  subroutine alloc_x_real_short(var, opt_global)
+  subroutine alloc_x_freal_short(var, opt_global)
 
      implicit none
 
-     real(mytype), allocatable, dimension(:, :, :) :: var
+     real(kind(0._real32)), allocatable, dimension(:, :, :) :: var
```

Comment:
I think you should only need to write `real(real32)` and similar for the double case below

Reply:
This will make the code easier to read. I will push it later.